### PR TITLE
feat: add user-by-connection query

### LINKED ISSUE #65: Feature Request: Twitch login query for v3 GQL
In v2 GQL, you could: Get a user by id, **login** or current authenticated user (@󠀀me).
Example:
https://old.7tv.app/users/brian6932 returns: https://old.7tv.app/users/60b04a7fad7fb4b50bd3a982

This is currently not possible with v3
https://7tv.app/users/brian6932 should return: https://7tv.app/users/60b04a7fad7fb4b50bd3a982

Problem being, this either generates unnecessary v2 REST requests, or forces people to use v2 GQL to query logins

### PR DESCRIPTION
**Pull request by Nerixyz**
This is an attempt to add a query that gets a user by its connection-id.

It's basically the `user` query from v2 mixed with the one from v3.

Fixes #65 

 _I can't test it but at least it builds_

### CODE REVIEW

File: internal/gql/v3/resolvers/query/query.users.go [hunk 1]
```diff
@@ -63,3 +64,27 @@ func (r *Resolver) Users(ctx context.Context, queryArg string, pageArg *int, lim
 
 	return result, err
 }
+
+func (r *Resolver) UserByConnection(ctx context.Context, connectionId string) (*model.User, error) {
+	user, err := r.Ctx.Inst().Query.Users(ctx, bson.M{"connections.id": strings.ToLower(connectionId)}).First()
```
Comment (AnatoleAM) [inline internal/gql/v3/resolvers/query/query.users.go:69]: This should be a dataloader, not a 1 for 1 query

File: internal/loaders/user.loader.go [hunk 2]
```diff
@@ -42,6 +42,13 @@ func userLoader[T comparable](ctx context.Context, x inst, keyName string) *data
 					case "username":
 						v, _ := utils.ToAny(u.Username).(T)
 						m[v] = u
+					case "connection.id":
+						for _, c := range u.Connections {
+							if c.ID != "" {
+								v, _ := utils.ToAny(c.ID).(T)
+								m[v] = u
+							}
+						}
```
Comment (Nerixyz) [inline internal/loaders/user.loader.go:51]: This assumes that connection-ids don't overlap.

**Comment by AnatoleAM** [comment]
Currently merged into https://github.com/SevenTV/API/pull/99 which adds REST support

**Comment by AnatoleAM** [comment]
Merged as part of #99. Thank you for your contribution

**Pull request closed without merging**